src/main/java/org/folio/fqm/repository/EntityTypeRepository.java

```diff
@@ -110,6 +137,63 @@ public void replaceEntityTypeDefinitions(List<EntityType> entityTypes) {
     });
   }
 
+  public Optional<EntityType> getCompositeEntityTypeDefinition(UUID entityTypeId) {
+    log.info("Getting COMPOSITE definition for entity type ID: {}", entityTypeId);
+    Field<String> definitionField = field("definition", String.class);
+
+    EntityType entityType = jooqContext
+      .select(definitionField)
+      .from(table(TABLE_NAME))
+      .where(field(ID_FIELD_NAME).eq(entityTypeId))
+      .fetchOptional(definitionField)
+      .map(this::unmarshallEntityType)
+      .map(currentEntityType -> {
+        String customFieldsEntityTypeId = currentEntityType.getCustomFieldEntityTypeId();
+        if (customFieldsEntityTypeId != null) {
+          currentEntityType.getColumns().addAll(fetchColumnNamesForCustomFields(UUID.fromString(customFieldsEntityTypeId)));
+        }
+        return currentEntityType;
+      }).orElseThrow();
+
+    // Build entity type from sources and joins
+//    var source1 = entityType.getSources().get(0);
+//    buildCompositeEntityTypeDefinition(entityType);
+
+    return Optional.of(entityType);
+  }
+
+  private void buildCompositeEntityTypeDefinition(EntityType entityType) {
+    System.out.println("BUILD COMPOSITE DEFINITION");
+    List<EntityTypeSourceJoin> entityTypeSourceJoins = new ArrayList<>();
+    List<EntityTypeSource> sources = entityType.getSources();
+    entityType.setFromClause("BUILD COMPOSITE DEFINITION");
+    StringBuilder fromClause = new StringBuilder();
+    for (int i = 0; i < sources.size(); i++) {
+      EntityTypeSource source = sources.get(i);
+      System.out.println("Building source: " + source.getAlias());
+      // TODO: below if-block checks if source is the first in the array, since there is nothing to join if it is.
+      //  But this needs to be done better
+      if (i == 0) { // TODO: think about replacing this with -- if (source.getJoin() == null)
+        fromClause.append(source.getAlias());
+      } else {
+        if (source.getType().equals("db")) {
```

Comment:
```suggestion
        if ("db".equals(source.getType())) {
```
This way, we handles nulls and avoid a potential NPE